### PR TITLE
Signin integration

### PR DESCRIPTION
**Pull request by Aliblas**
I have really had to experiment to make things execute at the correct timings, but loading on UserPage seems to work smoothly and consistently now.
Main changes are as follows:
GoogleAuth object initializer now takes a listener as a parameter, which will run when ever a user is detected.
In UserPage, GoogleAuth is initialized from the API external JavaScript, which guarantees it loads fully before initialization.
Initialization adds loadRecipes as a listener to currentUser, so recipes fill the page whenever a user signs in.
Sign in button has been added to topbar, and topbar has been added to UserPage.
Nothing is styled yet.

### CODE REVIEW

File: src/main/webapp/javascript/user-page.js [hunk 1]
```diff
@@ -6,9 +6,9 @@ const USER_RECIPES_DIV = "user-recipes";
 const PLANNER_RECIPES_DIV = "planner-recipes";
 const COOKBOOK_RECIPES_DIV = "cookbook-recipes";
 
-window.onload = function() {
-  onStart();
-  confirmUser().then(loadRecipes);
+// Can be passed to API onload to garuntee auth2 is loaded before execution.
```
Comment (Aliblas) [inline src/main/webapp/javascript/user-page.js]: I cannot spell.

**Review by isamitre — APPROVED** [review]

File: src/main/webapp/components/topbar.jsp [hunk 2]
```diff
@@ -2,7 +2,19 @@
...
+        <div id="profile-photo-container">
+        <img id="profile-photo" src=../images/default_profile.jpeg>
+        </div>
+        <div id="g-signin-container">
+        </div>
+        <div class="dropdown-content">
+        <a href="UserPage.jsp" id="user-page-anchor"></a>
+        <a href="#" onclick="signOut()">Sign out</a>
+        </div>
+    </div>
```
Comment (oturner24) [inline src/main/webapp/components/topbar.jsp]: nit: can you indent inside the divs?

File: src/main/webapp/javascript/user-auth.js [hunk 3]
```diff
@@ -18,16 +18,19 @@ function getIdToken() {
 /**
  * Loads the auth2 library for the g-signin API, and then initializes
  * the googleAuth object.
- * Passed as jquery parameter, executed by API script.
+ * Takes a listener args, to execute when a new user signs in.
```
Comment (oturner24) [inline src/main/webapp/javascript/user-auth.js]: nit: args -> arg ?

**Review by oturner24 — APPROVED** [review]
congrats on getting this working!

**Pull request merged**